pkg/image/admission/admission.go

```diff
@@ -0,0 +1,121 @@
+package admission
+
+import (
+	"fmt"
+	"io"
+
+	"github.com/golang/glog"
+
+	"github.com/openshift/origin/pkg/client"
+	oadmission "github.com/openshift/origin/pkg/cmd/server/admission"
+	imageapi "github.com/openshift/origin/pkg/image/api"
+
+	kadmission "k8s.io/kubernetes/pkg/admission"
+	kapi "k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/api/resource"
+	clientset "k8s.io/kubernetes/pkg/client/clientset_generated/internalclientset"
+	"k8s.io/kubernetes/pkg/runtime"
+	limitranger "k8s.io/kubernetes/plugin/pkg/admission/limitranger"
+)
+
+var pluginName string = "ImageLimitRange"
+
+func init() {
+	kadmission.RegisterPlugin(pluginName, func(client clientset.Interface, config io.Reader) (kadmission.Interface, error) {
+		return NewImageLimitRangerPlugin(client, config)
+	})
+}
+
+type imageLimitRangerPlugin struct {
+	*kadmission.Handler
+	osClient    client.Interface
+	LimitRanger kadmission.Interface
+}
+
+var _ = oadmission.WantsOpenshiftClient(&imageLimitRangerPlugin{})
+var _ = oadmission.Validator(&imageLimitRangerPlugin{})
+
+func NewImageLimitRangerPlugin(client clientset.Interface, config io.Reader) (kadmission.Interface, error) {
+	plugin := &imageLimitRangerPlugin{
+		Handler: kadmission.NewHandler(kadmission.Create),
+	}
+	limitRanger, err := limitranger.NewLimitRanger(client, plugin.imageLimitRanger)
+	if err != nil {
+		return nil, err
+	}
+	plugin.LimitRanger = limitRanger
+
+	return plugin, nil
+}
+
+func (a *imageLimitRangerPlugin) imageLimitRanger(limitRange *kapi.LimitRange, kind string, obj runtime.Object) error {
+	if !limitAppliesToImages(limitRange) {
+		return nil
+	}
+	ism, ok := obj.(*imageapi.ImageStreamMapping)
+	if !ok {
+		glog.V(4).Infof("obj with type %s was sent and could not be handled", kind)
+		return nil
+	}
+
+	img := ism.Image
+	if value, exists := img.Annotations[imageapi.ManagedByOpenShiftAnnotation]; !exists || value != "true" {
+		glog.V(5).Infof("image is not managed by openshift, ignoring")
+		return nil
+	}
+
+	imageSize := img.DockerImageMetadata.Size
```

Comment:
If some older registry is in use, the `DockerImageMetadata` may not be filled yet. Use `ImageWithMetadata()` to load it from manifest.